payment-engine/src/main/java/de/tillhub/paymentengine/opi/communication/OPIChannel0.kt

```diff
@@ -41,7 +40,7 @@ class OPIChannel0(
         coroutineScope.launch {
             try {
                 webSocket = socketFactory.createClientSocket(socketIp, socketPort)
-            } catch (e: Exception) {
+            } catch (e: IOException) {
```

Comment:
I think you need fallback catch of UnknownHostException as well here.

Reply:
UnknownHostException extends the IOException so it should cover both cases.